luigi/contrib/kubernetes.py

```diff
@@ -239,10 +249,19 @@ def signal_complete(self):
         pass
 
     def __get_pods(self):
-        pod_objs = Pod.objects(self.__kube_api, namespace=self.kubernetes_namespace) \
-            .filter(selector="job-name=" + self.uu_name) \
-            .response['items']
-        return [Pod(self.__kube_api, p) for p in pod_objs]
+        for _ in range(self.max_retrials_to_get_pods + 1):
+            pod_objs = Pod.objects(self.__kube_api, namespace=self.kubernetes_namespace) \
+                .filter(selector="job-name=" + self.uu_name) \
+                .response['items']
+            pods = [Pod(self.__kube_api, p) for p in pod_objs]
+
+            if pods:
+                break
+
+            # If pod was not returned, sleep to wait for pod to be created.
+            time.sleep(self.__POLL_TIME)
```

Comment:
if we're going to allow the number of retries to be configurable, shouldn't we also allow the wait time to be configurable as well? You may want to try a lot in a short period or only a few times over a longer duration